buildpacks/gradle/src/layers/gradle_home.rs

```diff
@@ -0,0 +1,90 @@
+use crate::{GradleBuildpack, GradleBuildpackError, GRADLE_TASK_NAME_HEROKU_START_DAEMON};
+use indoc::{formatdoc, indoc};
+use libcnb::build::BuildContext;
+use libcnb::data::layer_content_metadata::LayerTypes;
+use libcnb::generic::GenericMetadata;
+use libcnb::layer::{ExistingLayerStrategy, Layer, LayerData, LayerResult, LayerResultBuilder};
+use libcnb::layer_env::{LayerEnv, ModificationBehavior, Scope};
+use libcnb::Buildpack;
+use std::fs;
+use std::path::Path;
+
+pub(crate) struct GradleHomeLayer;
+
+impl Layer for GradleHomeLayer {
+    type Buildpack = GradleBuildpack;
+    type Metadata = GenericMetadata;
+
+    fn types(&self) -> LayerTypes {
+        LayerTypes {
+            launch: true,
+            build: true,
+            cache: true,
+        }
+    }
+
+    fn create(
+        &self,
+        _context: &BuildContext<Self::Buildpack>,
+        layer_path: &Path,
+    ) -> Result<LayerResult<Self::Metadata>, <Self::Buildpack as Buildpack>::Error> {
+        // https://docs.gradle.org/8.3/userguide/build_environment.html#sec:gradle_configuration_properties
+        fs::write(
+            layer_path.join("gradle.properties"),
+            indoc! {"
+                org.gradle.welcome=never
+                org.gradle.caching=true
+            "},
+        )
+        .map_err(GradleBuildpackError::WriteGradlePropertiesError)?;
+
+        // https://docs.gradle.org/8.3/userguide/init_scripts.html
+        fs::write(
+            layer_path.join("init.gradle.kts"),
+            formatdoc! {"
+                allprojects {{
+                    tasks.register(\"{task_name}\")
```

Comment:
why do you need to register this task with no body?

Reply:
Added in 0f54f01